### PR TITLE
Recommend 4MB Memory per partition

### PR DESCRIPTION
**Pull request by JakeSCahill**
Fixes redpanda-data/documentation-private#2001

### CODE REVIEW

File: modules/deploy/pages/deployment-option/self-hosted/manual/sizing.adoc [hunk 1]
```diff
@@ -14,6 +14,7 @@ In general, choose the number of nodes based on the following criteria, and add
 
 * Total network bandwidth must account for maximum simultaneous writes and reads, multiplied by the replication factor.
 * Have a minimum of 2 GB memory for each core, but more memory is better.
+* Have a minimum of 4 MB of memory for each topic partition. You can enforce this requirement in the tunable xref:reference:tunable-properties.adoc#topic_memory_per_partition[`topic_memory_per_partition` property].
```
Comment (daisukebe) [inline modules/deploy/pages/deployment-option/self-hosted/manual/sizing.adoc]: > 4 MB of memory for each topic partition.

See https://vectorizedio.atlassian.net/wiki/spaces/CORE/pages/99844097/Scale+Limits+in+Redpanda. It applies to partition replicas, therefore it should be

> 4 MB of memory for each topic partition replica.

File: modules/deploy/partials/requirements.adoc [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+== CPU and memory
+
+- Two physical, not virtual, cores for each {node}.
+
+- x86_64 (Westmere or newer) and AWS Graviton family processors are supported.
+
+- 2 GB or more of memory per core.
+
+- 4 MB of memory for each topic partition. You can enforce this requirement in the tunable xref:reference:tunable-properties.adoc#topic_memory_per_partition[`topic_memory_per_partition` property].
```
Comment (daisukebe) [inline modules/deploy/partials/requirements.adoc]: Same as above. The 4 MB requirement should apply to partition replicas.

**Review by Feediver1 — APPROVED** [review]
lgtm

**Review by daisukebe — APPROVED** [review]
lgtm, thank you!

**Pull request merged**